bot.py

```diff
@@ -1,160 +1,104 @@
+        if state == STATE_OPEN and persons is not None
+        else usernames[state]
+    )
+
+    for guild in client.guilds:
+        member = guild.get_member(client.user.id)
+        if member and member.nick != nick:
             try:
-                await client.user.edit(avatar=avatars[state])
+                await member.edit(nick=nick)
             except Exception as e:
-                logging.error(f'Failed to update avatar {state}: {e}')
+                logging.error(f'Failed to update nickname: {e}')
+
+    if state != current_state or avatar_update_pending:
+        try:
+            await client.user.edit(avatar=avatars[state])
+            avatar_update_pending = False
+        except Exception as e:
+            logging.error(f'Failed to update avatar to [{state}]: {e}')
+            avatar_update_pending = True
 
     current_state = state
     current_persons = persons
 
-    if confirmation_needed and state == 'closed':
-        confirmation_needed = False
-        pending_state = None
 
-
-# Fire every minute
 @tasks.loop(minutes=1)
 async def is_there_life_on_mars():
-    global confirmation_needed, confirmation_start_time, pending_state
-
-    if confirmation_needed:
-        time_elapsed = datetime.now() - confirmation_start_time
-        if time_elapsed >= timedelta(minutes=confirmation_delay_minutes):
-            logging.info(f"Confirmation period elapsed. Checking if space is still {pending_state}.")
-        else:
-            minutes_left = confirmation_delay_minutes - time_elapsed.total_seconds() / 60
-            logging.info(f"In confirmation period for {pending_state} state. {minutes_left:.1f} minutes left.")
-
-    logging.info('Checking the status')
+    logging.info('Checking the status...')
     try:
         response = requests.get(space_endpoint, timeout=10)
         spaceapi_json = response.json()
-        space_state = 'open' if spaceapi_json['state']['open'] else 'closed'
+
+        match (is_bot_enabled, spaceapi_json['state']['open']):
+            case (False, _):
+                space_state = STATE_UNKNOWN
+            case (True, True):
+                space_state = STATE_OPEN
+            case (True, False):
+                space_state = STATE_CLOSED
+
         people = 0
         try:
-            if ('sensors' in spaceapi_json and
-                    'people_now_present' in spaceapi_json['sensors'] and
-                    spaceapi_json['sensors']['people_now_present'] and
-                    isinstance(spaceapi_json['sensors']['people_now_present'], list)):
-
+            if (
+                'sensors' in spaceapi_json
+                and 'people_now_present' in spaceapi_json['sensors']
+                and spaceapi_json['sensors']['people_now_present']
+                and isinstance(spaceapi_json['sensors']['people_now_present'], list)
+            ):
                 people_data = spaceapi_json['sensors']['people_now_present'][0]
```

Comment:
Actually, this could be a sum of all the values in the list. Right now we have a single element, but it's a list after all.